src/picom.c

```diff
@@ -169,6 +170,26 @@ void cxinerama_upd_scrs(session_t *ps) {
 	free(xinerama_scrs);
 }
 
+static inline bool dpms_screen_is_off(xcb_dpms_info_reply_t *info) {
+	// state is a bool indicating whether dpms is enabled
+	return info->state && (info->power_level != XCB_DPMS_DPMS_MODE_ON);
+}
+
+void check_dpms_status(EV_P attr_unused, ev_timer *w, int revents attr_unused) {
+	auto ps = session_ptr(w, dpms_check_timer);
+	auto r = xcb_dpms_info_reply(ps->c, xcb_dpms_info(ps->c), NULL);
+	if (!r) {
+		log_fatal("Failed to query DPMS status.");
+		abort();
+	}
+	auto now_screen_is_off = dpms_screen_is_off(r);
+	if (ps->screen_is_off != now_screen_is_off) {
+		ps->screen_is_off = now_screen_is_off;
```

Comment:
```c
...
if (ps->screen_is_off != dpms_screen_is_off(r)) {
    ps->screen_is_off = !ps->screen_is_off
...
```

Reply:
hmm, i think mine is a bit less magical.